theme/icons.go

```diff
@@ -78,6 +83,29 @@ func (res *InvertedThemedResource) Invert() *ThemedResource {
 	return NewThemedResource(res.source, nil)
 }
 
+// ErrorThemedResource is a resource wrapper that will return a version of the resource with the main color changed
+// to indicate an error.
+type ErrorThemedResource struct {
+	source fyne.Resource
+}
+
+// NewErrorThemedResource creates a resource that adapts to the error color for the current theme.
+func NewErrorThemedResource(orig fyne.Resource) *ErrorThemedResource {
+	res := &ErrorThemedResource{source: orig}
+	return res
+}
+
+// Name returns the underlying resource name (used for caching).
+func (res *ErrorThemedResource) Name() string {
+	return "error-" + res.source.Name()
+}
+
+// Content returns the underlying content of the resource adapted to the current background color.
+func (res *ErrorThemedResource) Content() []byte {
+	clr := &color.NRGBA{0xf4, 0x43, 0x36, 0xff} // Should be current().ErrorColor() in the future
```

Comment:
nit: it is easy to forget this comment. Consider adding a "TODO" or "FIXME" prefix so it shows up in searches